### ISSUE TITLE
Forged local sender passed incoming on MX backup without requiring SMTP auth

### ISSUE THREAD

**Issue by andipfaff** [hmailserver/hmailserver#247]
I have s strange behaviour on my secondary hmalserver Installation. I floowed the instructions on https://www.hmailserver.com/documentation/latest/?page=howto_act_as_mx_backup but today an email from an IP in Argentina passed to my Inbox. Envelope Sender and Receiver was my local Email. The Primary server which Hosts the Domains and mailboxes denied the Sender wirth "Auth requeired":
"SMTPD"	2312	476234	"2018-01-19 22:01:17.288"	"181.24.155.202"	"SENT: 220 Welcome on our Mailserver"
"SMTPD"	2288	476234	"2018-01-19 22:01:17.772"	"181.24.155.202"	"RECEIVED: EHLO 181-24-155-202.speedy.com.ar"
"SMTPD"	2288	476234	"2018-01-19 22:01:17.772"	"181.24.155.202"	"SENT: 250-zuesrv01.cd-hotel.ch[nl]250-SIZE 30000000[nl]250-STARTTLS[nl]250-AUTH LOGIN PLAIN[nl]250 HELP"
"SMTPD"	2320	476234	"2018-01-19 22:01:18.428"	"181.24.155.202"	"RECEIVED: MAIL From:<andreas.pfaff@domain.de>"
"SMTPD"	2320	476234	"2018-01-19 22:01:18.834"	"181.24.155.202"	"SENT: 250 OK"
"SMTPD"	2312	476234	"2018-01-19 22:01:19.834"	"181.24.155.202"	"RECEIVED: RCPT To:<andreas.pfaff@domain.de>"
"SMTPD"	2312	476234	"2018-01-19 22:01:19.834"	"181.24.155.202"	"SENT: 530 SMTP authentication is required."

The Backup MX accepted the same Sender and message:
"SMTPD"	1212	100022	"2018-01-19 22:01:38.128"	"181.24.155.202"	"SENT: 220 Welcome on our Mailserver"
"SMTPD"	1060	100022	"2018-01-19 22:01:39.128"	"181.24.155.202"	"RECEIVED: EHLO 181-24-155-202.speedy.com.ar"
"SMTPD"	1060	100022	"2018-01-19 22:01:39.128"	"181.24.155.202"	"SENT: 250-zuesrv02[nl]250-SIZE 9000000[nl]250-STARTTLS[nl]250-AUTH LOGIN PLAIN[nl]250 HELP"
"SMTPD"	3644	100022	"2018-01-19 22:01:40.222"	"181.24.155.202"	"RECEIVED: MAIL From:<andreas.pfaff@domain.de>"
"SMTPD"	3644	100022	"2018-01-19 22:01:40.535"	"181.24.155.202"	"SENT: 250 OK"
"SMTPD"	1212	100022	"2018-01-19 22:01:41.957"	"181.24.155.202"	"RECEIVED: RCPT To:<andreas.pfaff@domain.de>"
"SMTPD"	1212	100022	"2018-01-19 22:01:41.957"	"181.24.155.202"	"SENT: 250 OK"
"SMTPD"	3644	100022	"2018-01-19 22:01:42.551"	"181.24.155.202"	"RECEIVED: DATA"
"SMTPD"	3644	100022	"2018-01-19 22:01:42.551"	"181.24.155.202"	"SENT: 354 OK, send."
"SMTPD"	984	100022	"2018-01-19 22:01:44.238"	"181.24.155.202"	"SENT: 250 Queued (1.664 seconds)"
"SMTPD"	3168	100022	"2018-01-19 22:01:44.863"	"181.24.155.202"	"RECEIVED: QUIT"
"SMTPD"	3168	100022	"2018-01-19 22:01:44.863"	"181.24.155.202"	"SENT: 221 goodbye"

This text in the description for the MX backup does not seem to be correct:
◦Select that sender should be treated as external and recipients as local. 
 When someone sends an email from the route domain name, they should be treated as external. If someone sends an email to the domain name, it should be treated as local. This has the effect that hMailServer will allow delivery to the primary server without SMTP auth because delivery from external to internal address does not require authentication!

When someone sends an email from the route domain Name (here Domain.de), they should be treated as INTERNAL, not external. In this case it would be treated as internal to internal which is configured to "authentication required".

![unbenannt-1](https://user-images.githubusercontent.com/4201912/35182884-a68b04f4-fdd4-11e7-9d2d-d4d9f30efa56.jpg)
![unbenannt-2](https://user-images.githubusercontent.com/4201912/35182885-a6b3b3d6-fdd4-11e7-8f57-df03c877ba31.jpg)
Or am I wrong?

**Comment by martinknafve** [comment]
This was a bit complicated :)

So in your case, both the sender and the recipient has the domain name domain.de. This is the same domain as in the route, so both the sender and recipient will match the route.

According to your route config, this means that:

- The sender will be treated as a remote/external email address
- The recipient will be treated as a local email address.

In the IP range, you've specifeid that no authentication should be required when a message is being sent from remote/external to a local address. So it seems correct to me that no auth is required?

It seems like you should configure the route so that "when sender matches route, treat sender as a local email address. This is the default configuration for routes and would have the effect that SMTP auth is required if attempting to send from domain.de on the backup server.

**Comment by andipfaff** [comment]
Exact. Sender adress and receipient address is the same and is in one of my local Domains. According to the recommendations on the hmailserver Website (!!!) on how to configure the Backup MX this recommandation allows faked Emails. "Sombody" in the Internet is able to send Emails to recipients in one of the Domains hosted on my Primary Mailserver.

I have reconfigured my Mailserver and will test it today.

The reason why I have opened this ticket is because somebody should correct the recommendation on the Website für the MX backup. If my tests will be OK there is no bug in the Software.
